tests/lib/mockMagic.py

```diff
@@ -9,3 +11,49 @@ def validate_config_for_device(self, device_mac, config):
             return return_validate_config_for_device['return_validate_intf']
         if config == validate_true['config']:
             return return_validate_config_for_device['return_validate_true']
+
+    def get_device_by_serial(self, device_serial):
+        """
+        mock get_device_by_serial to get device_facts
+        """
+        return device_data[0]
+
+    def get_image_bundle_by_name(self, name):
+        """
+        mock to get image_bundle
+        """
+        if device_data[0]['imageBundle'] != 'Invalid_bundle_name':
+            return image_bundle
+        else:
+            return None
+
+    def apply_image_to_element(self, image, element, name, id_type,
+                               create_task=True):
+        """
+        mock for apply_image_to_element
+        """
+        if 'imageBundleKeys' in image_bundle:
+            if image_bundle['imageBundleKeys']:
+                node_id = image_bundle['imageBundleKeys'][0]
+
+        if 'id' in image_bundle:
+            node_id = image_bundle['id']
+
+        elif 'key' in image_bundle:
+            node_id = image_bundle['key']
```

Comment:
will this condition ever be True based on your `image_bundle` mock data?

Reply:
no it won't be, removing this condition